llama_stack/cli/stack/_build.py

```diff
@@ -88,6 +88,45 @@ def run_stack_build_command(args: argparse.Namespace) -> None:
                 color="red",
             )
             sys.exit(1)
+    elif args.providers:
+        providers = dict()
+        try:
```

Comment:
this is too broad a `try-catch`. anything could have gone wrong anywhere. you should error out if there is no "=" in each argument directly.